cylc/flow/subprocctx.py

```diff
@@ -158,3 +158,17 @@ def get_signature(self):
         args = self.func_args + [
             "%s=%s" % (i, self.func_kwargs[i]) for i in skeys]
         return "%s(%s)" % (self.func_name, ", ".join([str(a) for a in args]))
+
+    def dump(self) -> str:
+        """Output for logging."""
+        return SubProcContext.__str__(self)
+
+    def __str__(self) -> str:
+        """
+        >>> str(SubFuncContext('label', 'my_func', [1, 2], {'a': 3}))
+        'my_func(1, 2, a=3):10.0'
+        """
+        return f"{self.get_signature()}:{self.intvl}"
```

Comment:
The previous `__str__()` implementation was removed in https://github.com/cylc/cylc-flow/pull/5617/files#r1254330367, but I've ensured the output from xtriggers still gets logged in the workflow log by adding a `dump` method and using that instead

Reply:
That was it! Great.